src/SOS/SOS.InstallHelper/InstallHelper.cs

```diff
@@ -0,0 +1,307 @@
+            {
+                existing = true;
+                bool markerFound = false;
+                foreach (string line in File.ReadAllLines(LLDBInitFile))
+                {
+                    if (line.Contains(InitFileEnd)) {
+                        markerFound = false;
+                        changed = true;
+                        continue;
+                    }
+                    if (!markerFound) {
+                        if (line.Contains(InitFileStart)) {
+                            markerFound = true;
+                            changed = true;
+                            continue;
+                        }
+                        lines.Add(line);
+                    }
+                }
+                if (markerFound) {
+                    throw new ArgumentException(".lldbinit file end marker not found");
+                }
+            }
+
+            // If configure (not remove), add the plugin load, etc. configuration between the start/end markers.
+            if (!remove)
+            {
+                lines.Add(InitFileStart);
+                string plugin = Path.Combine(InstallLocation, "libsosplugin");
+                string extension = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? ".dylib" : ".so";
+                lines.Add($"plugin load {plugin}{extension}");
+
+                if (EnableSymbolServer) {
+                    lines.Add(string.Format("setsymbolserver -ms"));
+                }
+                lines.Add(InitFileEnd);
+                changed = true;
+            }
+
+            // If there is anything to write, write the lldb init file
+            if (changed)
+            {
+                if (remove) {
+                    WriteLine("Reverting {0} file - LLDB will no longer load SOS at startup", LLDBInitFile);
+                }
+                else {
+                    WriteLine("{0} {1} file - LLDB will load SOS automatically at startup", existing ? "Updating existing" : "Creating new", LLDBInitFile);
+                }
+                RetryOperation($"Problem writing lldb init file {LLDBInitFile}", () => File.WriteAllLines(LLDBInitFile, lines.ToArray()));
+            }
+        }
+
+        /// <summary>
+        /// Retries any IO operation failures.
+        /// </summary>
+        /// <param name="errorMessage">text message or null (don't throw exception)</param>
+        /// <param name="operation">callback</param>
+        /// <exception cref="ArgumentException">errorMessage</exception>
+        private void RetryOperation(string errorMessage, Action operation)
+        {
+            Exception lastfailure = null;
+
+            for (int retry = 0; retry < 5; retry++)
+            {
+                try
+                {
+                    operation();
+                    return;
+                }
+                catch (Exception ex) when (ex is IOException)
+                {
+                    // Retry file copy possible recoverable exception
```

Comment:
You will probably want a sizable delay here to ensure that if there is a transient issue the next retry is unlikely to hit the same thing. I'd suggest sleeping 1 sec. We aren't copying enough files around for this to get that slow, even if every file took the maximum number of tries.